### ISSUE TITLE
Color of graph based on value

### ISSUE THREAD

**Issue by archerian** [humangeo/leaflet-dvf#53]
Is it possible to set the color of the bar/pie based on the value of a field? For example from the data below, set the color to Green if Score >= 95, Orange if  90 <= Score < 95 and Red if Score < 90? 

[{
        "Date" : "2014-09-01",
        "State" : "Alabama",
        "StateCode" : "AL",
        "Operators" : [{
                "Operator" : "CellServiceA",
                "Score" : 88.94
            }, {
                "Operator" : "CellServiceB",
                "Score" : 95.71
            }, {
                "Operator" : "CellServiceC",
                "Score" : 95.36
            }, {
                "Operator" : "CellServiceD",
                "Score" : 90.79
            }
        ]
    },...
]   

I tried something like below, but it applies the same color to all bars.

'Operators[Operator=CellServiceA].Score': {
                    fillColor: function (value) {
                            if(value >= 95)
                                return 'hsl(114,100%,50%)'; // Green
                            if(value >= 90)
                                return 'hsl(63,100%,50%)'; // Yellow
                            else
                                return 'hsl(0,100%,50%)'; // Red
                        }
                },
                'Operators[Operator=CellServiceA].Score': {
                    fillColor: function (value) {
                            if(value >= 95)
                                return 'hsl(114,100%,50%)'; // Green
                            if(value >= 90)
                                return 'hsl(63,100%,50%)'; // Yellow
                            else
                                return 'hsl(0,100%,50%)'; // Red
                        }
                },

**Comment by rickj33** [comment]
I needed the same functionality, So i started to extend the L.BarchartDataLayer and override the _getMarkerMethod, but there is an issue. In the L.ChartDatalyer and L.BarChartDataLayer the _getMarkerMethod override removes the record parameter. In the ChartDataLayer the _getLayer method calls the _getMarkerMethod but with out the record parameter even though the record object is passed into the _getLayer Method..
 It would be very helpful to add the record parameter back into the _getMarkerMethod of the ChartDataLayer and BarChartDataLayer 

The code i was looking at is in the 1.0Dev branch.

Thanks
Rick Jones

**Comment by sfairgrieve** [comment]
Good point.  Thanks for pointing this out.  The latest code should have this change in it.

**Comment by rickj33** [comment]
Thanks for responding so quickly.
